src/TParsec/Combinators.idr

```diff
@@ -76,128 +146,341 @@ andbind p q = MkParser $ \mlen, ts =>
+||| both parsers succeed. The parsed value of the second one is 
+||| discarded.
+|||
+||| Unindexed signature: `Parser a -> Parser b -> Parser a`
 land : (Monad mn, Alternative mn) =>
        All (Parser mn p a :-> Box (Parser mn p b) :-> Parser mn p a)
 land p q = map fst (and p q)
 
+||| Runs two parsers in succession but discard the first value.
+|||
+||| Runs two parsers one after the other, this parser succeeds if
+||| both parsers succeed. The parsed value of the first one is 
+||| discarded.
+|||
+||| Unindexed signature: `Parser a -> Parser b -> Parser b`
 rand : (Monad mn, Alternative mn) =>
        All (Parser mn p a :-> Box (Parser mn p b) :-> Parser mn p b)
 rand p q = map snd (and p q)
 
+||| Runs two parsers in succession, discard the second value, second parser might fail.
+|||
+||| Runs two parsers one after the other, this parser suceeds as
+||| long as the first parser succeeds, the second one might fail.
+||| The value parsed by the second parser is discarded and only 
+||| the value parsed by the first is returned.
+|||
+||| Unindexed signature: `Parser a -> Parser b -> Parser a`
 landopt : (Monad mn, Alternative mn) =>
           All (Parser mn p a :-> Box (Parser mn p b) :-> Parser mn p a)
 landopt p q = map fst (andopt p q)
 
+||| Runs two parsers in succession, discard the first value, second parser might fail.
+|||
+||| Runs two parsers one after the other, this parser suceeds as
+||| long as the first parser succeeds. The value parsed by the first
+||| parser is discarded. If the second parser fails the value 
+||| `Nothing` is returned.
+|||
+||| Unindexed signature: `Parser a -> Parser b -> Parser (Maybe b)`
 randopt : (Monad mn, Alternative mn) =>
           All (Parser mn p a :-> Box (Parser mn p b) :-> Parser mn p (Maybe b))
 randopt p q = map snd (andopt p q)
 
+||| Runs two parsers in succession, discard the second value, first parser might fail.
+|||
+||| Runs two parsers one after the other, this parser suceeds as
+||| long as the second parser succeeds. The value parsed by the second
+||| parser is discarded. If the first parser fails the value 
+||| `Nothing` is returned.
+|||
+||| Unindexed signature: `Parser a -> Parser b -> Parser (Maybe a)`
 loptand : (Monad mn, Alternative mn) =>
           All (Parser mn p a :-> Parser mn p b :-> Parser mn p (Maybe a))
 loptand p q = map fst (optand p q)
 
+
+||| Runs two parsers in succession, discard the first value, first parser might fail.
+||| 
+||| Runs two parsers one after the other, this parser suceeds as
+||| long as the second parser succeeds, the first one might fail.
+||| The value parsed by the first parser is discarded and only 
+||| the value parsed by the second is returned.
+|||
+||| Unindexed signature: `Parser a -> Parser b -> Parser b`
 roptand : (Monad mn, Alternative mn) =>
           All (Parser mn p a :-> Parser mn p b :-> Parser mn p b)
 roptand p q = map snd (optand p q)
 
+||| Attempts to parse using two different parsers.
+|||
```

Comment:
May be worth to add "and remember where the success came from"

Reply:
I tried that but it made the first sentence too long. I've modified both the description sentence and the paragraph below to accommodate for your feedback, can you tell me if this is acceptable?

Reply:
:+1: